applications/nrf5340_audio/src/bluetooth/bt_management/scanning/bt_mgmt_scan_for_conn.c

```diff
@@ -156,6 +162,62 @@ static bool device_name_check(struct bt_data *data, void *user_data)
 	return true;
 }
 
+static bool csip_found(struct bt_data *data, void *user_data)
+{
+	int ret;
+	bt_addr_le_t *addr = user_data;
+	struct bt_conn *conn;
+	char addr_string[BT_ADDR_LE_STR_LEN];
+
+	if (bt_csip_set_coordinator_is_set_member(sirk_local, data)) {
+		/* Check if the device is still connected due to waiting for ACL timeout */
+		struct bt_conn_info info;
+		struct bt_conn *existing_conn = bt_conn_lookup_addr_le(BT_ID_DEFAULT, addr);
+
+		if (existing_conn != NULL) {
+			ret = bt_conn_get_info(existing_conn, &info);
+			if (ret == 0 && info.state == BT_CONN_STATE_CONNECTED) {
+				LOG_DBG("Trying to connect to an already connected conn");
+				bt_conn_unref(existing_conn);
+				return false;
+			}
+
+			/* Unref is needed due to bt_conn_lookup */
+			bt_conn_unref(existing_conn);
+		}
+		LOG_DBG("Coordinated set device found");
+		sirk_local = NULL;
+
+		bt_le_scan_cb_unregister(&scan_callback);
+		cb_registered = false;
+
+		ret = bt_le_scan_stop();
```

Comment:
Should we stop the scan before doing `bt_le_scan_cb_unregister` ? What happens if we get a scan result between line 191 and 194?

Reply:
It shouldn't matter. If we get a scan result and there is cb then nothing will happen